### ISSUE TITLE
[bug] Can't send stop message to Fibaro Roller Shutter

### ISSUE THREAD

**Issue by foxy82** [OpenZWave/Zwave2Mqtt#123]
**Version**

Build/Run method
- [x] Docker
- [ ] PKG
- [ ] Manually built (git clone - npm install - npm run build ) 

Zwave2Mqtt version: 1.2.3
Openzwave Version: 1.4.1

**Describe the bug**
Can't send stop command to Fibaro Roller Shutter 2.

When I use Home Assistants built in zwave integration (via OpenZwave 1.4.1) it will send the following to stop the window whilst it is opening.

2019-09-20 20:36:27.875 Info, Node022, SwitchMultilevel::StopLevelChange - Stopping the level change
2019-09-20 20:36:27.875 Detail, Node022, Queuing (Send) SwitchMultilevelCmd_StopLevelChange (Node=22): 0x01, 0x09, 0x00, 0x13, 0x16, 0x02, 0x26, 0x05, 0x25, 0x88, 0x7f
2019-09-20 20:36:27.875 Detail, Node022, Queuing (Send) SwitchBinaryCmd_Get (Node=22): 0x01, 0x09, 0x00, 0x13, 0x16, 0x02, 0x25, 0x02, 0x25, 0x89, 0x7a
2019-09-20 20:36:27.876 Detail, Node022, Queuing (Send) SwitchMultilevelCmd_Get (Node=22): 0x01, 0x09, 0x00, 0x13, 0x16, 0x02, 0x26, 0x02, 0x25, 0x8a, 0x7a
2019-09-20 20:36:27.876 Detail, Node022, Queuing (Send) SensorMultilevelCmd_Get (Node=22): 0x01, 0x09, 0x00, 0x13, 0x16, 0x02, 0x31, 0x04, 0x25, 0x8b, 0x6a
2019-09-20 20:36:27.876 Detail, Node022, Queuing (Send) MeterCmd_Get (Node=22): 0x01, 0x0a, 0x00, 0x13, 0x16, 0x03, 0x32, 0x01, 0x00, 0x25, 0x8c, 0x69
2019-09-20 20:36:27.877 Detail, 
2019-09-20 20:36:27.877 Info, Node022, Sending (Send) message (Callback ID=0x88, Expected Reply=0x13) - SwitchMultilevelCmd_StopLevelChange (Node=22): 0x01, 0x09, 0x00, 0x13, 0x16, 0x02, 0x26, 0x05, 0x25, 0x88, 0x7f`
`

I can't find a way to send this from zwave2mqtt here is what I see in the web page: 

![image](https://user-images.githubusercontent.com/7756020/65355244-66cbb580-dbe9-11e9-8460-658b95f8ac5e.png)

**Comment by robertsLando** [comment]
@foxy82 Never seen `StopLevelChange` method in openzwave-shared apis, @Fishwaldo how does I call that method?

**Comment by Fishwaldo** [comment]
It’s a ValueButton exposed as a ValueID.

**Comment by robertsLando** [comment]
Ok so @foxy82 did you try to use bright/dim buttons?

**Comment by foxy82** [comment]
Hi,

So the bright / dim buttons work to make the window open to 100% and close will fully close the window. 

The operation I'm trying to do is hit the open button - wait till the window is as open as I want it and the hit stop. This worked with the Home Assistant OpenZwave integration and is where the log comes from. I cut off the bit where I hit open but can provide logs if that helps?

**Comment by robertsLando** [comment]
@foxy82 I think that there should be another button to stop but I don't know where it is :( did you try to send close while opening? does it close fully or stops it?

**Comment by foxy82** [comment]
If I hit "Dim" (close) whilst it is opening then it closes the whole way.

**Comment by robertsLando** [comment]
@foxy82 That's strange, maybe it is not present in your device configuration? Can you tell me the name of device?

**Comment by foxy82** [comment]
It is the Fibaro Roller Shuter 2 (https://manuals.fibaro.com/roller-shutter-2/)

**Comment by foxy82** [comment]
Interestingly in my zwcfg using Home Assistant it looks like this:

`        <Node id="22" name="Table Vent" location="" basic="4" generic="17" specific="6" type="Motor Control Class B" listening="true" frequentListening="false" beaming="true" routing="true" max_baud_rate="40000" version="4" query_stage="Complete">
                <Manufacturer id="10f" name="FIBARO System">
                        <Product type="302" id="1000" name="FGRM222 Roller Shutter Controller 2" />
                </Manufacturer>
                <CommandClasses>
                        <CommandClass id="32" name="COMMAND_CLASS_BASIC" version="1" request_flags="4" mapping="38">
                                <Instance index="1" />
                        </CommandClass>
                        <CommandClass id="37" name="COMMAND_CLASS_SWITCH_BINARY" version="1" request_flags="4" innif="true">
                                <Instance index="1" />
                                <Value type="bool" genre="user" instance="1" index="0" label="Switch" units="" read_only="false" write_only="false" verify_changes="false" poll_intensity="0" min="0" max="0" value="True" />
`
Where as in the zwave2mqtt docker it is:

`        <Node id="22" name="" location="" basic="4" generic="17" specific="6" type="Routing Slave" listening="true" frequentListening="false" beaming="true" routing="true" max_baud_rate="40000" version="4" query_stage="ManufacturerSpecific2">
                <Manufacturer id="0" name="">
                        <Product type="0" id="0" name="" />
                </Manufacturer>
                <CommandClasses>
                        <CommandClass id="37" name="COMMAND_CLASS_SWITCH_BINARY" version="1" request_flags="4" innif="true">
                                <Instance index="1" />
                                <Value type="bool" genre="user" instance="1" index="0" label="Switch" units="" read_only="false" write_only="false" verify_changes="false" poll_intensity="0" min="0" max="0" value="False" />
                        </CommandClass>
`

**Comment by foxy82** [comment]
Some more info - I decided to backup my zwave2mqtt zwcfg file and copy the one generated by Home Assistant and now these buttons appear:

![image](https://user-images.githubusercontent.com/7756020/65418987-93541d00-ddf5-11e9-9e18-0ee379ed27dc.png)

Still I don't have a Stop. I'm not sure what the on/off toggle is meant to do however when it is opening it stops the window however when it is closing it doesn't and when the window isn't moving it will start it opening.

**Comment by robertsLando** [comment]
@Fishwaldo Is home assistant using a different configuration file? Actually its configuration is https://github.com/OpenZWave/open-zwave/blob/master/config/fibaro/fgrm222.xml is it missing something?

**Comment by petergebruers** [comment]
Yes they use a different _source_ for configuration files, but afaik the configuration contents are the same and hass is not missing anything, this is a pretty old device btw.

Hass forked the 1.4 repo, the hass installer refers to: https://github.com/home-assistant/open-zwave/blob/hass/config/fibaro/fgrm222.xml

What you linked, is "master" so 1.6

You can spot the difference by looking at the first line: `<Product xmlns='https://github.com/OpenZWave/open-zwave' Revision="5">` -> different names space and has "Revision" tag -> this is for 1.6

**Comment by robertsLando** [comment]
@petergebruers Thanks for pointing this out! Is that fork keep updated by hass so? Should I use that as device db? Actually in docker conatiner I'm fetching devices db from master branch of openzwave maybe all problems are related to this?

**Comment by petergebruers** [comment]
I've reread my statement and I admit I took a few shortcuts and it does not convey my message well... Sorry about that, let me try again...

The origin of the config files should match the version of OZW, when you build product X from source and that source is based on "master" you'll have OZW 1.6 with matching file.

Likewise, you could still check out the "1.4" branch (of the canonical repository) and build the OZW library and use the "config" subfolder - still everything will match.

Trouble starts when users come from another system and start copying over config files... Which is why I've mentioned the HASS situation, it applies to the Z-Wave Integration that has existed for several years now. If users install that Z-Wave part, it will pull in a fork of "pyozw" and that one in turn builds OZW (1.4-ish because it is a fork) from <https://github.com/home-assistant/open-zwave> - So when users start copying files around trouble indeed may start. Now OZW 1.6 also has that "revision" tag and fetches updates from a server that uses the "master" config files, something the 1.4 build does not do.

On my main system (a Mac) I debug and test doing "make install" and many of my (test) programs get their files from the system location. OZW 1.6 will tell you that, eg

```
Loading Localization File /usr/local/etc/openzwave/Localization.xml With Revision 8
```

Hass users, especially when they run the "docker" variants, might have set an explicit directory with a copy of the config dir, fo example to add a device.

Although the "Z-Wave Integration" does not have anything to do with Zwave2Mqtt I think it is best to be aware of this situation, I am sometimes active on their Z-Wave Discord channel and from time to time people talk about moving, migrating and copying stuff between systems and also between the Integration and Zwave2Mqtt

**Comment by petergebruers** [comment]
> and maybe this could cause problems based on what you say.

The formats are indeed incompatible. 

> I did this because I thought that by installing 1.4 version devices db would match the device db of that version (that is very old, about 5/6 years ago)

I am not sure what you mean by "installing" the 1.4 database... Afaik the "config" folder always has been part of the repo...

And the 1.4 branch certainly does have more recent commits in the config folder...

See:

https://github.com/OpenZWave/open-zwave/tree/1.4/config

Last commit may 2019 eg "Merge pull request #1780 from derekbelrose/Kwikset_SmartCode_914"

But @Fishwaldo and I are no longer accepting updates of that 1.4 branch so don't expect more changes...

> I don't understand if when you build 1.4 the device db is compiled from master or what else

Yes, but a few months ago the "Dev" (aka 1.5) branch was merged into master, turning master into "OZW 1.6"... Then a 1.6 tag was added to mark that start. After that, conversion was run by @Fishwaldo (not all files yet iirc) to add metadata, images and so on.

**Comment by Fishwaldo** [comment]
Just another note - in 1.6 OZW will download the latest config files so there is no need to “grab” the latest database.

**Comment by robertsLando** [comment]
Ok perfect I have fixed the docker file as follow: https://github.com/robertsLando/Zwave2Mqtt-docker/blob/master/Dockerfile#L33 now binaries and devices database are fetched and compiled from 1.4 branch. Thanks for pointing this out @petergebruers! I didn't know that there was different devices database for each ozw version. 

@Fishwaldo What do you mean? Does ozw 1.6 automatically refresh its devices database everytime it is started or just on the installation process?

**Comment by petergebruers** [comment]
When OZW starts, it checks the revision of several files and downloads the latest revision. I think it still does make sense to provide users with a config folder, not all may like to allow http access to download the files so I would still include a config folder primed with the contents of the git repo... When someone commits a new revision to the master repo, that revision becomes available over http. I don't know the exact details of that process, between the web server and github, Fishwaldo maintains that part. 

You can look at the current device database by browsing to <http://openzwave.com/device-database>

**Comment by foxy82** [comment]
Thanks for all the work. As I understand it using the wrong config made the zwcfg file incorrect and changing to use the correct version will fix that.

However as I copied the 1.4 version of cfg over from running the same zwave network under home assistant and still didn't get a stop button is this actually going to fix the issue or will the issue still be there just wit better config?

**Comment by robertsLando** [comment]
Remove the device and test it with latest version from docker and let me know. Sorry I can’t guarantee that this will fix the issue because I woukd need the same device you have to test it

Daniel

**Comment by foxy82** [comment]
> Ok perfect I have fixed the docker file as follow: https://github.com/robertsLando/Zwave2Mqtt-docker/blob/master/Dockerfile#L33 now binaries and devices database are fetched and compiled from 1.4 branch. Thanks for pointing this out @petergebruers! I didn't know that there was different devices database for each ozw version.

How do I use this? Sorry I'm a newbie with docker.

**Comment by foxy82** [comment]
Hi, so I've migrated to using zwave2mqtt via hass.io add-ons. This was upgraded last night which I think includes the change above? I uninstalled the addin completely (which I assume blows away the XML) however I still get a layout with Bright/Dim buttons instead of the Inc/Dec one I get when using the cfg created by the Home assistant direct integration with zwave.

**Comment by robertsLando** [comment]
Nope you have to remove and add back the nodes to check if they take the correct xml file

Daniel

**Comment by foxy82** [comment]
By remove do you mean exclude from the network and include it again or can it be done any other way?

**Comment by robertsLando** [comment]
@foxy82 I mean removeNode from the network and add it back again yes

**Comment by Flight777** [comment]
I can confirm that this is not fixed unfortunately, the Roller Shutters 2 are still detected with the Bright/Dim and also autodiscovery for hass is set to lights instead of the covers.

@foxy82 did you find any solution yet?

**Comment by robertsLando** [comment]
@foxy82 @Flight777 try to use Ozw 1.6 to see if this has been fixed

**Comment by foxy82** [comment]
How can we use 1.6 is there any documentation for that?

**Comment by robertsLando** [comment]
If you are using docker simply use the new tag ‘-dev’ check dockerhub for more info

Daniel

**Comment by kintrupf** [comment]
Is there anything new to this? I have the same problem. I'm using yesterdays latest Docker image of ZWave2mqtt (version 2.2.0). On the "User" page of the Fibaro there's the switch item, and the four buttons labeled "Inc", "Dec", "Open" and "Close", but "Stop" is still missing.
I'm coming from openHAB, and the ZWave binding there did support a stop button out of the box for this device (but had other problems, so I switched to Z2M).

BTW, is there an mqtt topic which directly corresponds to the "Open" and "Close" buttons? Right now I've only managed to move the shutters by posting 0 or 99  (or anything in between) to the topic "zwave/<myShutterName>/switch_multilevel/level".

**Comment by robertsLando** [comment]
@Fishwaldo

**Comment by Kroontje** [comment]
@robertsLando any news? Same issue with a flush shutter from Qubino: ZMNHCDx Flush Shutter (Qubino (Goap)). The device is correctly discovered by Z2M, but no function to open, close or stop with commands.

**Comment by robertsLando** [comment]
@Kroontje What version are you using?

**Comment by Kroontje** [comment]
@robertsLando version: v3.0.1

**Comment by robertsLando** [comment]
For this kind of issues @Fishwaldo can help
